nodewatcher/nodewatcher.py

```diff
@@ -154,35 +169,46 @@ def main():
     log.info("nodewatcher startup")
     instance_id = getInstanceId()
     hostname = getHostname()
-    region, asg_name, scheduler, proxy_config = getConfig(instance_id)
+    region, asg_name, scheduler, proxy_config, idle_time = getConfig(instance_id)
+
 
     s = loadSchedulerModule(scheduler)
 
+    if os.path.isfile(_NW_IDLETIME_FILE):
+        data = json.loads(open(_NW_IDLETIME_FILE).read())
+    else:
+        data = {'idle_time': 0}
+
+    atexit.register(saveIdleTime, data)
     while True:
         time.sleep(60)
-        ec2_conn = boto3.resource('ec2', region_name=region, config=proxy_config)
         asg_conn = boto3.client('autoscaling', region_name=region, config=proxy_config)
-        hour_percentile = getHourPercentile(instance_id, ec2_conn)
-        log.info('Percent of hour used: %d' % hour_percentile)
-
-        if hour_percentile < 95:
-            continue
 
         jobs = getJobs(s, hostname)
-        if jobs == True:
+        if jobs:
             log.info('Instance has active jobs.')
+            data['idle_time'] = 0
         else:
             if maintainSize(asg_name, asg_conn):
                 continue
             # avoid race condition by locking and verifying
             lockHost(s, hostname)
             jobs = getJobs(s, hostname)
-            if jobs == True:
+            if jobs:
                 log.info('Instance actually has active jobs.')
+                data['idle_time'] = 0
                 lockHost(s, hostname, unlock=True)
                 continue
             else:
-                selfTerminate(asg_name, asg_conn, instance_id)
+                data['idle_time'] += 1
+                log.info('Instance %s has no job for the past %s minute(s)' % (instance_id, data['idle_time']))
+                if data['idle_time'] >= idle_time:
```

Comment:
it's confusing to use the same label/ var name for what is the *current_idletime* and the *scaledown_idletime/threshold/period*